src/datasource.ts

```diff
@@ -101,7 +103,9 @@ export class DataSource extends DataSourceApi<
           ? calculateQueryInterval(target.interval, range!.from.valueOf(), range!.to.valueOf())
           : undefined;
 
-      let queryFrom = range!.from;
+      // create new moment object to not mutate the original grafana object with startOf() to not change
+      // the grafana graph as this would change the timeframe for all the following queries
+      let queryFrom = moment(range!.from.valueOf());
```

Comment:
Wasn't there a reason, why we wanted to mutate original `range.from` ? I remember, we discuss that for half hour+ in one of techsyncs ... otherwise we would use local `let queryFrom` from beginning ....

Reply:
Yes, the reason was to adjust the timeframe of the graph through modifying the range.from with the startOf() function to be able to display the first datapoint, otherwise it would still be there but outside the graph's boundaries. 

I added the range.from() modification again, but only after all the queries were fetched and data was transformed. This way we send the starttimes for all the queries correctly but also adapt the timeframe of the graph to display the first datapoint.